W-13784959-image based mouse clicks (#220)
mulesoft/docs-rpa: rpa-recorder/6.4/modules/ROOT/pages/user-interface.adoc

[%header,cols="10,20,30,10"]
|===
|Icon |Name | Description | Hotkey

> | |Image-based mouse clicks | Add recording with the <<Image-Based Mouse Clicks>> feature. | `F3`

Reviewer: This is missing the icon. Also, the description is not parallel to the other table descriptions.

Author: Thank you! The icon was pending availability of the latest RPA Recorder version. I'll be adding that shortly.

> |image:image-based-mouse-clicks-icon.png[Image-Based Mouse Clicks icon, 20%, 20%]|Image-based mouse clicks | Adds a recording with the <<Image-Based Mouse Clicks>> feature. | `F3`

Reviewer 2: suggestion: how about "Adds a the Image-Based Mouse Clicks to the current recording." ?

When a recording state changes, a toast notification appears indicating current state. The notification stays for two seconds and disappears automatically.

== Image-Based Mouse Clicks

> Use image-based mouse clicks at any point during a normal recording session to record mouse clicks on a UI element that otherwise cannot be recorded properly. This feature guides you through the mouse click process. An image on the screen and mouse clicks must be captured, and the mouse clicks are recorded relative to the captured image.

Reviewer 2: suggestion: remove the word "normal" here and below. I guess that it was from our PM, and the word is commonly used in German. There is just one "recording session" (neither normal nor abnormal) with two possible activity types - "Bot Task" or "Web Task". IBMC is available in both, just as "Add screenshot" is available in both. That's what this sentence is trying to say.
suggestion: how about changing "_This feature guides you through the mouse click process_." to "_This feature guides you through the capture of image and mouse clicks._ " 
praise: the rest is good and clear!

The feature also checks your display settings automatically while recording to ensure the feature records correctly.

> This feature will result in _Search Pattern_ and _Mouse Action_ Action Steps when the workflow is generated in RPA Builder from the recording.

Reviewer 2: praise: very good and short, brings it to the point

When using this feature:

* The Recording Toolbar is inactive.
* Keyboard events are not recorded.

=== Recommended Display Settings

Apply the following recommended display settings to capture image-based mouse clicks:

* Single monitor. 
+

> Although the image-based mouse clicks feature supports multiple monitors, that setting limits the capturing to the primary display for Bot tasks and the web page for Web tasks.

Reviewer 2: suggestion: correct "that setting limits the capturing to..." to "the capturing is limited to...". The display settings don't limit or restrict the recording (as of now).

* Display scaling of 100%. If the scaling is different, then there is a risk that the captured image will not be found by the workflow generated in RPA Builder from the recording.

> To check your display settings before recording, go to *Windows Settings > Display*. If your *Scale and layout* or *Multiple displays* settings are different than what's recommended when recording, a warning message shows each time you record an image-based mouse click, and you must confirm the message.

Reviewer 2: issue: if user wants to get rid of that warning, it is not enough to change the Windows Setting for Display, but also to restart the RPA Recorder completely (i.e. close the application and open it again). Please add a sentence saying that when changing the Windows Settings, user will need to restart the RPA Recorder for the changed Display settings to have any effect on the recording.

=== Using Image-Based Mouse Clicks

To capture image-based mouse clicks:

. Start a normal recording process.
. While recording, press *F3* to start recording image-based mouse clicks, or press the *Image-Based Mouse Clicks* icon on the Recording Toolbar.
. Capture an image by selecting it on the screen, accept or retake the image, and then set a mouse click, which the Recorder marks with a red dot. Record additional mouse clicks, such as double-mouse clicks, if desired.
. Click *Finish*. This generates all relevant data for RPA Builder to create the _Search Pattern_ and _Mouse Action_ Action Steps for the recording.

The process documentation and PDF generated by the Recorder include the recording data. Once you finish using image-based mouse clicks, you can continue with the normal recording process.

When recording image-based mouse clicks in a Web Task:

* Scrolling or zooming in the displayed web page or moving or resizing the browser window results in inaccurate relative clicks in the recording.

> * Any subsequent click after navigating to another web page can cause inaccuracies in the recording.

Reviewer 2: suggestion: replace "Any subsequent click after navigating to another web page can cause inaccuracies in the recording." with "Navigating to a different web page during capturing the Image-Based Mouse Clicks (e.g. by clicking on a link) can cause inaccuracies in the recording of the subsequent relative mouse clicks."

When a recording state changes, a toast notification appears indicating current state. The notification stays for two seconds and disappears automatically.

== Image-Based Mouse Clicks

> Use image-based mouse clicks to record mouse clicks on a UI element that otherwise cannot be recorded properly. The feature guides you through the mouse click process. An image on the screen and mouse clicks must be captured, and the mouse clicks are recorded relative to the captured image.

Reviewer: ```suggestion
Use image-based mouse clicks at any point during a normal recording session to record mouse clicks on a UI element that otherwise cannot be recorded properly. This feature guides you through the mouse click process. An image on the screen and mouse clicks must be captured, and the mouse clicks are recorded relative to the captured image.
```

Reviewer: Do you need the word "normal"? Also, can you make the last sentence active? I am not completely understanding the sentence. Do you capture the screen before using the feature and then have it record the clicks?

> Use this feature at any point during a normal recording session. The feature also checks your display settings automatically while recording to ensure the feature records correctly.

Reviewer: ```suggestion
The feature also checks your display settings automatically while recording to ensure the feature records correctly.
```

> This feature will result in _Search Pattern_ and _Mouse Action_ Action Steps when the workflow is generated in RPA Builder from the recording.

Reviewer: I am confused. Here you say it results in _Search Pattern_ and _Mouse Action_ Action Steps and then you say it is available for _Search Pattern_ and _Mouse Action_ Action Steps.

Reviewer 2: "results in Search Pattern and Mouse Action Action Steps" is correct

> This feature is available for _Search Pattern_ and _Mouse Action_ Action Steps.

Reviewer 2: I suggest to remove "This feature is available for _Search Pattern_ and _Mouse Action_ Action Steps.". The statement above is sufficient, in my opinion: "This feature will result in _Search Pattern_ and _Mouse Action_ Action Steps when the workflow is generated in RPA Builder from the recording."

When using this feature:

* The Recording Toolbar is inactive.
* Keyboard events are not recorded.

=== Recommended Display Settings

> Apply the following displays settings to capture image-based mouse clicks:

Reviewer: ```suggestion
Apply the following recommended display settings to capture image-based mouse clicks:
```

> * Single monitor. Multiple monitors are supported, but the capturing will be limited to the primary display only for Bot tasks and to the web page for Web tasks.

Reviewer: ```suggestion
* Single monitor. 
+
Although the image-based mouse clicks feature supports multiple monitors, that setting limits the capturing to the primary display for Bot tasks and the web page for Web tasks.
```

Reviewer: Global: Kim wanted us to put extra bullet text on a new line after a horizontal space to make the text more readable. (I marked this for the first bullet item.)

> * Display scaling of 100%. If the scaling is different, then there is a risk that the captured image will not be found by the workflow generated in RPA Builder from the recording.

Reviewer: Make active

> To check your display settings before recording, go to *Windows Settings > Display*. If your *Scale and layout* or *Multiple displays* settings are different than what's recommended when recording, a warning message shows each time you record an image-based mouse click, and you will need to confirm the message each time it shows.

Reviewer: ```suggestion
To check your display settings before recording, go to *Windows Settings > Display*. If your *Scale and layout* or *Multiple displays* settings are different than what's recommended when recording, a warning message shows each time you record an image-based mouse click, and you must confirm the message.
```

Reviewer: Do you need to say that you must confirm the message? I think that's obvious and doesn't have to be documented.

=== Using Image-Based Mouse Clicks

To capture image-based mouse clicks:

. Start a normal recording process.

> . While recording, press *F3* to start recording image-based mouse clicks, or press the *Image-Based Mouse Clicks* button on the Recording Toolbar. This opens the wizard for capturing image-based mouse clicks.

Reviewer: ```suggestion
. While recording, press *F3* to start recording image-based mouse clicks, or press the *Image-Based Mouse Clicks* button on the Recording Toolbar.
```

> . Capture an image by selecting it on the screen, accept or retake it, and then set a mouse click. The mouse click is marked with a red dot.

Reviewer: ```suggestion
. Capture an image by selecting it on the screen, accept or retake the image, and then set a mouse click, which the Recorder marks with a red dot.
```

> . Record additional mouse clicks, such as double-mouse clicks, if desired. All additional mouse clicks are recorded relative to the captured pattern and are marked with a red dot.

Reviewer 2: Just to be clear here: all mouse clicks are recorded relatively to the captured image pattern. There is no difference between the first mouse click and the following ones. In other words, user may record 1 to n relative mouse clicks, relative to the selected image pattern. It's not that the first one is a "main" one and others are additional, they are all recorded and treated equally.

> . Record additional mouse clicks, such as double-mouse clicks, if desired. All additional mouse clicks are recorded relative to the captured pattern and are marked with a red dot.

Reviewer: Make active.

. Click *Finish*. This generates all relevant data for RPA Builder to create the _Search Pattern_ and _Mouse Action_ Action Steps for the recording.

To use image-based mouse clicks for the _Search Pattern_ Action Step:

. Start a normal recording process.
. While recording, press *F3* to start recording image-based mouse clicks, or press the *Image-based mouse clicks* button on the Recording Toolbar. This opens the wizard for the Action Step.

> . Select *Capture pattern*.

Reviewer 2: There is no "Capture pattern" button. Instead, the area where use can select a pattern is "frozen" and the cursor changes to a cross, so user can capture a pattern image immediately, without further clicks. This area is the entire primary display when recording Bot Tasks or web page area when recording Web Tasks.

> . Draw a rectangular box to capture the desired pattern. The box shows a red frame around the capture. The search area is set to full-screen by default.

Reviewer: I don't think you need the second sentence.

Reviewer 2: Agree. The "search area" is an optional detail from the Search Pattern Action Step in the RPA Builder, we don't need to mention it in the RPA Recorder manual.

> . Select *Accept selection & continue* to continue with the relative mouse click action. All additional mouse clicks are recorded relative to the captured pattern and are marked with a red dot.

Reviewer 2: We shortened the button label from "*Accept selection & continue*" to just "*Accept*", please change the text. Thanks!

. Click *Finish*. This generates all relevant data for RPA Builder to create the _Search Pattern_ Actions Steps for the recording.

> The data for the recording is included in the process documentation and the PDF generated by the Recorder. Once you have finished using the image-based mouse clicks feature, you can continue with the normal recording process.

Reviewer: Make active.

> The data for the recording is included in the process documentation and the PDF generated by the Recorder. Once you have finished using the image-based mouse clicks feature, you can continue with the normal recording process.

Reviewer: ```suggestion
The process documentation and PDF generated by the Recorder include the recording data. Once you finish using image-based mouse clicks, you can continue with the normal recording process.
```

[NOTE]

> If you close the Action Step wizard by clicking the *X*, the image-based mouse clicks are not kept.

Reviewer 2: Which dialog exactly is meant by "Action Step wizard"?
Action Steps are a concept from RPA Builder, there should not be any Action Step UI in the Recorder.

When recording image-based mouse clicks in a Web Task:

> * Scrolling or zooming in the displayed web page, or moving or resizing the browser window, results in inaccurate relative clicks in the recording.

Reviewer: ```suggestion
* Scrolling or zooming in the displayed web page or moving or resizing the browser window results in inaccurate relative clicks in the recording.
```

> * Any subsequent click after navigating to another web page, such as by clicking a link, may cause inaccuracies in the recording.

Reviewer: ```suggestion
* Any subsequent click after navigating to another web page can cause inaccuracies in the recording.
```
